Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -0,0 +1,40 @@
+class JavaRequirement
+  cask "java"
+
+  env do
+    env_java_common
+    java_home = Pathname.new(@java_home)
+    if (java_home/"include").exist? # Oracle JVM
```

Comment:
Suggestion: just make this `env_oracle_jdk || env_apple` on line 6 and change `env_oracle_jdk` and skip `next unless @java_home` in favour of changes below.